### PR TITLE
Add claude 3 / anthropic with cost calculation + convenience fixes. gpt-engineer is coauthor

### PR DESCRIPTION
**Pull request by AntonOsika**

### CODE REVIEW

**Review by ErikBjare — APPROVED** [review]
LGTM, just need to fix the failing test

**Comment by viborc** [comment]
@AntonOsika, can you check these tests so we can proceed with this?

File: gpt_engineer/core/default/steps.py [hunk 1]
```diff
@@ -327,14 +329,16 @@ def salvage_correct_hunks(
...
+    for _, diff in diffs.items():
         # if diff is a new file, validation and correction is unnecessary
         if not diff.is_new_file():
             problems = diff.validate_and_correct(
                 file_to_lines_dict(files_dict[diff.filename_pre])
             )
             error_message.extend(problems)
     files_dict = apply_diffs(diffs, files_dict)
+    memory.log(IMPROVE_LOG_FILE, "\n\n".join(x.pretty_repr() for x in messages))
+    memory.log(DIFF_LOG_FILE, "\n\n".join(error_message))
```
Comment (ErikBjare) [inline gpt_engineer/core/default/steps.py:341]: undefined variable
```suggestion
```

**Comment by ErikBjare** [comment]
Fixed the remaining issues (afaik). Merging as soon as tests pass.

**Comment by ErikBjare** [comment]
Looks like there were more issues and broken tests. Not quite as easy as I thought to clean up.

File: gpt_engineer/core/ai.py [hunk 2]
```diff
@@ -110,7 +110,7 @@ def __init__(
 
         logger.debug(f"Using model {self.model_name}")
 
-    def start(self, system: str, user: str, step_name: str) -> List[Message]:
+    def start(self, system: str, user: str, *, step_name: str) -> List[Message]:
```
Comment (ErikBjare) [inline gpt_engineer/core/ai.py:113]: What was the idea with this `*`?

**Comment by ErikBjare** [comment]
Reviewing this PR has made me realize that we need to typecheck the codebase in CI: https://github.com/gpt-engineer-org/gpt-engineer/pull/1070

**Pull request closed without merging**